torch/testing/_internal/distributed/rpc/dist_examples/parameter_server.py

```diff
@@ -0,0 +1,108 @@
+# https://github.com/pytorch/examples/blob/master/distributed/rpc/batch/parameter_server.py
+
+import threading
+from datetime import datetime
+
+import torch
+import torch.distributed.rpc as rpc
+import torch.nn as nn
+from torch import optim
+
+batch_size = 20
+image_w = 64
+image_h = 64
+num_classes = 30
+batch_update_size = 3
+num_batches = 6
+
+
+def timed_log(text):
+    print(f"{datetime.now().strftime('%H:%M:%S')} {text}")
+
+
+class BatchUpdateParameterServer(object):
+
+    def __init__(self, batch_update_size=batch_update_size):
+        self.model = nn.Linear(100, num_classes)
+        self.lock = threading.Lock()
+        self.future_model = torch.futures.Future()
+        self.batch_update_size = batch_update_size
+        self.curr_update_size = 0
+        self.optimizer = optim.SGD(self.model.parameters(), lr=0.001, momentum=0.9)
+        for p in self.model.parameters():
+            p.grad = torch.zeros_like(p)
+
+    def get_model(self):
+        return self.model
+
+    @staticmethod
+    @rpc.functions.async_execution
+    def update_and_fetch_model(ps_rref, grads):
+        self = ps_rref.local_value()
+        timed_log(f"PS got {self.curr_update_size}/{batch_update_size} updates")
+        for p, g in zip(self.model.parameters(), grads):
+            p.grad += g
+        with self.lock:
+            self.curr_update_size += 1
+            fut = self.future_model
+
+            if self.curr_update_size >= self.batch_update_size:
+                for p in self.model.parameters():
+                    p.grad /= self.batch_update_size
+                self.curr_update_size = 0
+                self.optimizer.step()
+                self.optimizer.zero_grad()
+                fut.set_result(self.model)
+                timed_log("PS updated model")
+                self.future_model = torch.futures.Future()
+
+        return fut
+
+
+class Trainer(object):
+
+    def __init__(self, ps_rref):
+        self.ps_rref = ps_rref
+        self.loss_fn = nn.MSELoss()
+        self.one_hot_indices = torch.LongTensor(batch_size) \
```

Comment:
Since we are no longer using a ResNet model, we can use a simpler loss function and simpler labels here. This the link below can serve as an example. You don't need to use the same model. The simple `Linear` layer looks good to me. 

https://github.com/pytorch/pytorch/blob/172ed51a17e70aabfbdc096491fde79755de9d08/test/distributed/test_c10d.py#L1928-L1940
https://github.com/pytorch/pytorch/blob/172ed51a17e70aabfbdc096491fde79755de9d08/test/distributed/test_c10d.py#L2073